crates/sui-types/src/messages.rs

```diff
@@ -1353,3 +1356,23 @@ impl ConsensusTransaction {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum ExecuteTransactionRequestType {
+    ImmediateReturn,
+    WaitForTxCert(Duration),
+    WaitForEffectsCert(Duration),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ExecuteTransactionRequest {
+    pub transaction: Transaction,
+    pub request_type: ExecuteTransactionRequestType,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum ExecuteTransactionResponse {
+    ImmediateReturn,
+    TxCert(Box<CertifiedTransaction>),
+    EffectsCert(Box<(CertifiedTransaction, TransactionEffects)>),
```

Comment:
This is a bit misleadingly named, isn't it? The TransactionEffects here is unsigned, so the client can't use this as a proof of finality. An effects cert would actually be a `TransactionEffectsEnvelope<AuthorityQuorumSignInfo>`, right?

FWIW I may look at forming effects certificates in full node, since I'm already going to be doing the work of waiting for finality before syncing a cert. I say "may" because fullnode can get away with merely proving finality to itself by observed 2f+1 idential effects digests - forming an fx cert involves further fetching 2f+1 signed effects.

Reply:
The plan here is to make it return TransactionEffectsEnvelope<AuthorityQuorumSignInfo> eventually.
Currently since authority_aggregator only returns effects, here I am leaving it as it is. I plan to send future PRs to actually make authority_aggregator return effects certs.